crates/examples/infra/mod.rs

```diff
@@ -921,33 +922,7 @@ pub async fn main_entry_point<
     .await
     .expect("failed to get config");
 
-    let builder_task = match run_config.builder {
-        BuilderType::External => None,
-        BuilderType::Random => {
-            let (builder_task, builder_url) =
-                <RandomBuilderImplementation as TestBuilderImplementation<TYPES>>::start(
-                    run_config.config.num_nodes_with_stake.into(),
-                    run_config.random_builder.clone().unwrap_or_default(),
-                )
-                .await;
-
-            run_config.config.builder_url = builder_url;
-
-            builder_task
-        }
-        BuilderType::Simple => {
-            let (builder_task, builder_url) =
-                <SimpleBuilderImplementation as TestBuilderImplementation<TYPES>>::start(
-                    run_config.config.num_nodes_with_stake.into(),
-                    (),
-                )
-                .await;
-
-            run_config.config.builder_url = builder_url;
-
-            builder_task
-        }
-    };
+    let builder_task = initialize_builder(&mut run_config, &args, &orchestrator_client).await;
```

Comment:
Is it possible to learn builder address and assign it to `args.builder_address` before this line?